### PR TITLE
fix(output) : aws_rds_cluster shows $0 monthly cost but it should show "cost depends on usage" 

### PR DESCRIPTION
**Pull request by NeverovDS**
Issue #738

### CODE REVIEW

**Comment by NeverovDS** [comment]
@ekifel ready to check

**Comment by ekifel** [comment]
@NeverovDS this PR is required in rebase to fix the tests

File: internal/providers/terraform/aws/rds_cluster.go [hunk 1]
```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
...
-			snapShotStorageSizeGB = decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float())
-		}
+			snapShotStorageSizeGB = decimalPtr(decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float()))
+
+			totalBackupStorageGB := calculateBackupStorage(*snapShotStorageSizeGB, backupStorageRetention)
+			if totalBackupStorageGB.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBackupStorageCostComponent(region, &totalBackupStorageGB, databaseEngine))
+			}
+		} else {
+			var unknown *decimal.Decimal
```
Comment (ekifel) [inline internal/providers/terraform/aws/rds_cluster.go:75]: Looks like we can just add `totalBackupStorage` to `var` and use it after `if` block
```suggestion
			totalBackupStorageGB = calculateBackupStorage(*snapShotStorageSizeGB, backupStorageRetention)
			}
			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, totalBackupStorageGB, databaseEngine))
		} 
```

File: internal/providers/terraform/aws/rds_cluster.go [hunk 2]
```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
...
 
-		backtrackWindowHours := decimal.Zero
-		if u != nil && u.Get("backtrack_window_hrs").Exists() {
-			backtrackWindowHours = decimal.NewFromInt(u.Get("backtrack_window_hrs").Int())
-		}
+			if totalBacktrackChangeRecords.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBacktrackCostComponent(region, &totalBacktrackChangeRecords))
+			}
+		} else {
+			var unknown *decimal.Decimal
```
Comment (ekifel) [inline internal/providers/terraform/aws/rds_cluster.go:94]: The same thing as above, `totalBacktrackChangeRecords` in `var` and use it after `if` block.
```suggestion
		} 
		costComponents = append(costComponents, auroraBacktrackCostComponent(region, totalBacktrackChangeRecords))
```

File: internal/providers/terraform/aws/rds_cluster.go [hunk 3]
```diff
@@ -126,29 +147,25 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
...
 	}
 
-	readRequestsPerSecond := decimal.Zero
-	if u != nil && u.Get("read_requests_per_sec").Exists() {
-		readRequestsPerSecond = decimal.NewFromInt(u.Get("read_requests_per_sec").Int())
+	if u != nil && u.Get("write_requests_per_sec").Exists() && u.Get("read_requests_per_sec").Exists() {
+		writeRequestsPerSecond = decimalPtr(decimal.NewFromInt(u.Get("write_requests_per_sec").Int()))
+		readRequestsPerSecond = decimalPtr(decimal.NewFromInt(u.Get("read_requests_per_sec").Int()))
+		monthlyIORequestsD := calculateIORequests(*readRequestsPerSecond, *writeRequestsPerSecond)
+		monthlyIORequests = &monthlyIORequestsD
```
Comment (ekifel) [inline internal/providers/terraform/aws/rds_cluster.go:160]: monthlyIORequestsD variable is not necessary here
```suggestion
              monthlyIORequests = decimalPtr(calculateIORequests(*readRequestsPerSecond, *writeRequestsPerSecond))
```

File: internal/providers/terraform/aws/rds_cluster.go [hunk 4]
```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
...
 
-	snapshotExportSizeGB := decimal.Zero
+	var snapshotExportSizeGB *decimal.Decimal
 	if u != nil && u.Get("snapshot_export_size_gb").Exists() {
-		snapshotExportSizeGB = decimal.NewFromFloat(u.Get("snapshot_export_size_gb").Float())
-	}
+		snapshotExportSizeGB = decimalPtr(decimal.NewFromFloat(u.Get("snapshot_export_size_gb").Float()))
+
+		if snapshotExportSizeGB.GreaterThan(decimal.Zero) {
+			costComponents = append(costComponents, &schema.CostComponent{
```
Comment (ekifel) [inline internal/providers/terraform/aws/rds_cluster.go:105]: We can use only one `costComponents = append...` here, just take out this part of code outside of `if`, then `else` block not needed

**Review by ekifel — APPROVED** [review]
Thanks @NeverovDS! Almost done, some minor changes

**Comment by ekifel** [comment]
@alikhajeh1 LGTM!

**Pull request merged**